straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,130 @@
+# A new start signal can occur only after the previous busy instance ended.
+# 1ms (1e6 ns) - minimum busy veto length, or until the board clears its memory
+    
+# DDC10 High Energy Veto:
+# 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+# in XENONnT it will be calibrated based on the length of large S2 SE tails
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.24'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U10'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = straxen.n_hard_aqmon_start
+        aqmon_channel_names = ('sum_wf','m_veto_sync',
+                               'hev_stop', 'hev_start', 'he_stop','he_start','busy_stop', 'busy_start')
+        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
+        
+        # Keeping ch_range as class attribute for now, maybe will need it for other features
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
+        self.veto_names = ['busy_','he_','hev_']
+        return self.channel_map
+    
+    def get_window_size(self):
+        # Give a very wide window
+        return (self.config['max_veto_gap']*100)
+    
+    def compute(self, aqmon_hits): 
+        hits = aqmon_hits
+        
+        # Make a dict to populate later
+        res = dict()
+        
+        for i, veto in enumerate(self.veto_names):
+            veto_hits = channel_select(hits, self.channel_map[veto + 'stop'], self.channel_map[veto + 'start'])
+            vetos = self.merge_vetos(veto_hits, gap = self.config['max_veto_gap'],\
+                                           dtype = strax.time_fields, t = 0)
+            
+            # Check that we found a veto interval and update the resulting dict
```

Comment:
Not a major comment but usually we'd make a large buffer of the dtype you need and then clip until the length you actually added things. Might be somewhat faster.

Would look something like:
```python
buff_len = 10_000
res = np.zeros(buff_len, dtype = self.dtype)
n_res = 0

if len(vetos):
   res[n_res]['time'] = vetos['time']
   ... <same for others>
   n_res += 1
   if n_res > buff_len:
      raise ValueError('ran out of buffer') 
res[:n_res']
```

Reply:
Cheers, though, I think I've seen straxen plugins implementations without buffering first. Also, these data are so tiny compared to anything else that we will process, so I don't think it will make a difference. If you think that this is a show stopper, let me know, and I'd redo it again. Otherwise, I would keep at as it is..

Reply:
sure